### PR TITLE
george / WALL-265 / Extract all_payment_agent_list variable from cashier PaymentAgentStore to reusable hook in @deriv/hooks 

### PR DESCRIPTION
**Pull request by heorhi-deriv**
## Changes:

Please include a summary of the change and which issue is fixed below:
- replace `all_payment_agent_list` with reusable hook
- replace `is_payment_agent_list_visible_in_onboarding` with reusable hook

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: packages/cashier/src/components/cashier-onboarding/cashier-onboarding.tsx [hunk 1]
```diff
@@ -66,6 +66,10 @@ const CashierOnboarding = observer(({ setSideNotes }: TCashierOnboardingProps) =
     const is_currency_banner_visible =
         (!is_crypto && !can_change_fiat_currency) || (is_crypto && available_crypto_currencies.length > 0);
 
+    const { data: all_payment_agent_list, isLoading: is_loading } = usePaymentAgentList();
+
+    const is_payment_agent_visible_in_onboarding = Boolean(all_payment_agent_list?.length);
```
Comment (rostislav-deriv) [inline packages/cashier/src/components/cashier-onboarding/cashier-onboarding.tsx:71]: Why not use `!!all_payment_agent_list?.length`

File: packages/hooks/src/usePaymentAgentList.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { useFetch } from '@deriv/api';
+import { useStore } from '@deriv/stores';
+
+const usePaymentAgentList = (currency?: string) => {
+    const { client } = useStore();
+    const { residence } = client;
+
+    const { data, ...rest } = useFetch('paymentagent_list', {
+        payload: { paymentagent_list: residence, currency },
+        options: { enabled: Boolean(residence) },
```
Comment (rostislav-deriv) [inline packages/hooks/src/usePaymentAgentList.ts:10]: Same here

**Review by rostislav-deriv — APPROVED** [review]

File: packages/cashier/src/components/cashier-onboarding/cashier-onboarding.tsx (hunk 1 above)
Comment (rostislav-deriv) [reply packages/cashier/src/components/cashier-onboarding/cashier-onboarding.tsx:71]: `Boolean(...)` is more readable tho

File: packages/components/stories/icon/icons.js
Comment (farzin-deriv) [inline packages/components/stories/icon/icons.js:1]: @heorhi-deriv I believe you need to revert this file changes 👀

Comment (nijil-deriv) [reply packages/components/stories/icon/icons.js:1]: Yes, please.

**Review by farzin-deriv — APPROVED** [review]

**Pull request merged**